subsys/bluetooth/host/id.c

```diff
@@ -773,10 +773,28 @@ void bt_id_add(struct bt_keys *keys)
 		goto done;
 	}
 
-	err = hci_id_add(keys->id, &keys->addr, keys->irk.val);
-	if (err) {
-		BT_ERR("Failed to add IRK to controller");
-		goto done;
+	/*
+	 * Core Spec 5.3, Vol 4, Part E, Section 7.8.38:
+	 * If an entry already exists in the resolving list with the
+	 * same four parameter values, the Controller shall either reject
+	 * the command or not add the device to the resolving list again
+	 * and return success. If the command is rejected then the
+	 * error code Invalid HCI Command Parameters (0x12) should be used.
+	 */
+	bool dev_added = false;
+
+	if (bt_dev.le.rl_entries > 1) {
```

Comment:
can't you just put those two conditions in the same `if` statement, and remove the `dev_added` var ?